packages/core/src/client.ts

```diff
@@ -270,11 +272,9 @@ export class Client extends AsyncEventEmitter<MappedEvents> {
 					chunkCount: data.chunk_count,
 				};
 
-				if (data.chunk_index >= data.chunk_count - 1) {
-					break;
-				} else {
-					timer = createTimer();
-				}
+				if (data.chunk_index >= data.chunk_count - 1) break;
+
+				timer = createTimer(controller, timeout);
```

Comment:
Oof, this shows a bittt of an issue with our current code, which is that we keep creating timers and _not_ clearing out previous ones... We should probs solve that too